nix/modules/flake-parts/pre-commit.nix

```diff
@@ -7,13 +7,21 @@
   perSystem = { config, ... }: {
     pre-commit.settings = {
       hooks = {
-        nixpkgs-fmt.enable = true;
+        nixpkgs-fmt = {
+          enable = true;
+          # cabal2nix, nixpkgs-fmt both modifies default.nix, hence exlude fmt.
+          excludes = [ "cabal.nix" ];
```

Comment:
```suggestion
          excludes = [ config.pre-commit.settings.hooks.cabal2nix.settings.output_filename ];
```

(Or you could use the `config` from pre-commit module)

Reply:
@srid Should I use it here as well ?
https://github.com/srid/haskell-template/pull/146/files#diff-3f93ac0b8de1127777e463e9420d69914f8140f7d6b50c85ce51054ef09da127R33

Reply:
Although this was not needed. But I put it just for example.

Reply:
I am talking about this.

<img width="815" alt="Screenshot 2024-12-11 at 12 56 31 AM" src="https://github.com/user-attachments/assets/4e4d0357-0124-41d0-992e-849fbb6a9ce9">

Reply:
How can you use it there? It'd be a cyclic dependency.

Reply:
This is for `haskell-flake`. There are in total 3 places where I am using `cabal.nix`

1. While setting output_filename for `cabal2nix` pre-commit
2. While setting exclude for `nixpkgs-fmt`
3. For haskell-flake to refer to the cached`cabal2nix` file

In this particular case I was talking about 3.

ideally we can set at one place use it at all other places.

Reply:
Oh, right, you can just reuse the module option (3) in both the palces (1 & 2). Good idea.